mesonbuild/compilers/mixins/clike.py

```diff
@@ -1090,34 +1090,34 @@ def tuple_key(x: str) -> T.Tuple[int, ...]:
         return sorted(filtered, key=tuple_key, reverse=True)
 
     @classmethod
-    def _get_trials_from_pattern(cls, pattern: str, directory: str, libname: str) -> T.List[Path]:
-        f = Path(directory) / pattern.format(libname)
+    def _get_trials_from_pattern(cls, pattern: str, directory: str, libname: str) -> T.List[str]:
+        f = os.path.join(directory, pattern.format(libname))
```

Comment:
This change overall looks good. There was never any real reason to use pathlib here especially since we spent a lot of time transforming strings into Path objects just to reconvert them back into strings.